ic-cdk/src/call.rs

```diff
@@ -251,82 +232,321 @@ impl<'m, 'a> Call<'m, 'a> {
         self
     }
 
-    /// Sets the call to have a guaranteed response.
+    /// Changes the timeout for bounded response waiting.
     ///
-    /// If [`change_timeout`](Self::change_timeout) is invoked after this method,
-    /// the call will instead be set with best-effort responses.
-    pub fn with_guaranteed_response(mut self) -> Self {
-        self.timeout_seconds = None;
-        self
-    }
-
-    /// Sets the timeout for best-effort responses.
+    /// [`Call::bounded_wait`] defaults to a 10-second timeout.
     ///
-    /// If not set, the call defaults to a 10-second timeout.
     /// If invoked multiple times, the last value takes effect.
-    /// If [`with_guaranteed_response`](Self::with_guaranteed_response) is invoked after this method,
-    /// the timeout will be ignored.
+    ///
+    /// If invoked on an unbounded call constructed by [`Call::unbounded_wait`] , it will panic.
     ///
     /// # Note
     ///
-    /// A timeout of 0 second **DOES NOT** mean guranteed response.
-    /// The call would most likely time out (result in a `SysUnknown` reject).
+    /// A timeout of 0 second **DOES NOT** mean unbounded response waiting.
+    /// The call would most likely time out (result in a [`SysUnknown`](RejectCode::SysUnknown) reject).
     /// Unless it's a call to the canister on the same subnet,
     /// and the execution manages to schedule both the request and the response in the same round.
     ///
-    /// To make the call with a guaranteed response,
-    /// use the [`with_guaranteed_response`](Self::with_guaranteed_response) method.
+    /// To boundedly wait for response, use the  [`Call::bounded_wait`] constructor instead.
```

Comment:
(github won't let this be a suggestion) this should probably be moved up to right under the panic warning, which should probably be in a # Panics section. And the warning should probably be in a `<div class="warning">` section

Reply:
Sure, I will add a "# Panics" section.

For the last sentence, I meant to tell the users what should they do if they really need **unbounded** response waiting call. And I incorrectly referred to the `bounded_wait` method. I will correct it.